ee/indexeddb/indexeddb.go

```diff
@@ -0,0 +1,184 @@
+// Package indexeddb provides the ability to query an IndexedDB and parse the objects it contains.
+package indexeddb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/kolide/goleveldb/leveldb"
+	"github.com/kolide/goleveldb/leveldb/opt"
+	"github.com/kolide/launcher/ee/agent"
+)
+
+// maxObjectStoresToCheck is the number of indices for object stores we will check
+// before declaring failure to find the given object store. We cannot look up
+// object stores by their names, only by their IDs -- so we have to iterate through
+// up to maxObjectStoresToCheck IDs to find the desired store. We assume there are
+// fewer than 100 object stores in a given database.
+const maxObjectStoresToCheck = 100
+
+// QueryIndexeddbObjectStore queries the indexeddb at the given location `dbLocation`,
+// returning all objects in the given database that live in the given object store.
+func QueryIndexeddbObjectStore(dbLocation string, dbName string, objectStoreName string) ([]map[string]any, error) {
+	// If Chrome is open, we won't be able to open the db. So, copy it to a temporary location first.
+	tempDbCopyLocation, err := copyIndexeddb(dbLocation)
+	if err != nil {
+		if tempDbCopyLocation != "" {
+			_ = os.RemoveAll(tempDbCopyLocation)
+		}
+		return nil, fmt.Errorf("unable to copy db: %w", err)
+	}
+	defer os.RemoveAll(tempDbCopyLocation)
+
+	opts := &opt.Options{
+		Comparer:               newChromeComparer(),
+		DisableSeeksCompaction: true, // no need to perform compaction
+		Strict:                 opt.StrictAll,
+	}
+	db, err := leveldb.OpenFile(tempDbCopyLocation, opts)
+	if err != nil {
+		// Perform recover in case we missed something while copying
+		db, err = leveldb.RecoverFile(tempDbCopyLocation, opts)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("opening db: %w", err)
+	}
+	defer db.Close()
+
+	// Get the database ID from the name
+	databaseNameKey, err := databaseIdKey(dbLocation, dbName)
+	if err != nil {
+		return nil, fmt.Errorf("getting database id key: %w", err)
+	}
+	databaseIdRaw, err := db.Get(databaseNameKey, nil)
+	if err != nil {
+		return nil, fmt.Errorf("querying for database id: %w", err)
+	}
+	databaseId, _ := binary.Uvarint(databaseIdRaw)
+
+	// We can't query for the object store ID by its name -- we have to query each ID to get its name,
+	// and check against that. Object store indices start at 1.
+	var objectStoreId uint64
+	for i := 1; i < maxObjectStoresToCheck; i++ {
```

Comment:
I don't care either way, but since index is at one, shouldn't be be `<=` for the max number to to check. Vs `<` if it's the max index. But maybe I'm really saying that the plural "object stores" `maxObjectStoresToCheck` makes me think it's a count, not not a named index.

Reply:
Should `i` be a uint64, not an int? (Would it let you drop the casts later)

Reply:
Renamed the const to `maxNumberOfObjectStoresToCheck` -- I think that makes it clearer? Agreed on both counts re: `<=` and making `i` a uint64; updated!